### ISSUE TITLE
`machinectl shell --uid=somelocaluser` doesn't work

### ISSUE THREAD

**Issue by DSpeichert** [systemd/systemd#3472]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)
### systemd version the issue has been seen with

> 230
### Used distribution

> Arch Linux
### In case of bug report: Expected behaviour you didn't see

> Until v229 it was possible to run `machinectl shell --uid=somelocaluser`. The documentation doesn't indicate it was intentionally changed.
### In case of bug report: Unexpected behaviour you saw

> `machinectl shell` with specifying `--uid` stopped working
### In case of bug report: Steps to reproduce the problem

```
# machinectl shell --uid=somelocaluser
Failed to get shell PTY: No machine '--uid=somelocaluser' known
```

**Comment by evverx** [comment]
It was intentionally changed: https://github.com/systemd/systemd/issues/2420

**Comment by evverx** [comment]
BTW:
I'm not sure. Does [InterfaceStabilityPromise](https://www.freedesktop.org/wiki/Software/systemd/InterfaceStabilityPromise/) include `machinectl`?

> We will make sure that scripts invoking these commands will continue to work with future versions of systemd. 

...

> These are our intentions, but if in the future there are very good reasons to change or get rid of an interface we have listed above as stable, then we might take the liberty to do so, despite this promise. However, if we do this, then we'll do our best to provide a smooth and reasonably long transition phase.

**Comment by poettering** [comment]
Hmm, I figure we should still handle the switches up until the container name is specified, so there is a bug to fix here. i.e. currently machinectl shell works like this:

```
machinectl [switches A] shell [switches B] <container> [switches C]
```

The old code made machinectl handle all three sets of switches A, B and C. The new code makes machinectl handle the sets of switches A, doesn't recognized any switches B (assumes they are already the container name), and pases switches C to the container. And I figure this should be changed so that machinectl handles the sets of switches A and B, and passes on switches C to the invoked shell.

**Comment by poettering** [comment]
Fix waiting in #3572.
